### PR TITLE
Add secure credentials for API client tests

### PR DESCRIPTION
**Pull request by Joerger**
Completes an old TODO I left to use secure credentials in api client tests.

I also did some minor refactors to the mock server and affected tests.

This is also needed in https://github.com/gravitational/teleport/pull/30203 since Per-rpc MFA creds require a secure client connection.

I recommend reviewing commit by commit as the overall diff is hard to follow.

### CODE REVIEW

**Review by smallinsky — APPROVED** [review]

**Comment by Joerger** [comment]
@ibeckermayer Friendly ping to review

**Review by ibeckermayer — APPROVED** [review]
